provisor/provisor.py

```diff
@@ -140,23 +140,9 @@ def next_uid(self):
         continue
       return u
 
-  """ Returns the next gid for use """
-  def next_gid(self):
-    gids = []
-    results = self.con.search_s(self.group_base, ldap.SCOPE_ONELEVEL, '(objectClass=*)', ("gidNumber",), 0)
-    for r in results:
-      for attrs in r[1]:
-        gids.append(int(r[1][attrs][0]))
-    gids.sort()
-    for g in range(self.min_uid,self.max_uid,1):
-      if g in gids or g in self.excluded_uids:
-        continue
-      return g
 
-
-  def add_group(self, groupname, gid=-1):
-    if gid < 0:
-      self.next_gid()
+  def add_group(self, groupname, gid):
+    assert(gid>0)
```

Comment:
On Wed, Feb 24, 2016 at 05:27:04PM -0800, daurnimator wrote:
> >  
> > -
> > -  def add_group(self, groupname, gid=-1):
> > -    if gid < 0:
> > -      self.next_gid()
> > +  def add_group(self, groupname, gid):
> > +    assert(gid>0)
> 
> should possibly be `gid > 1000` to prevent system ones being used.

Good catch.  It can/should probably even be “gid > 3000” given that's where the curent user range starts.